src/query/graphite/ts/series.go

```diff
@@ -0,0 +1,645 @@
+package ts
+
+import (
+	"errors"
+	"math"
+	"time"
+
+	"github.com/m3db/m3/src/query/graphite/context"
+	"github.com/m3db/m3/src/query/graphite/stats"
+)
+
+var (
+	// ErrRangeIsInvalid is returned when attempting to slice Series with invalid range
+	// endpoints (begin is beyond end).
+	ErrRangeIsInvalid = errors.New("requested range is invalid")
+)
+
+// An AggregationFunc combines two data values at a given point.
+type AggregationFunc func(a, b float64) float64
+
+// A Series is the public interface to a block of timeseries values.  Each block has a start time,
+// a logical number of steps, and a step size indicating the number of milliseconds represented by each point.
+type Series struct {
+	name      string
+	startTime time.Time
+	vals      Values
+	ctx       context.Context
+
+	// The Specification is the path that was used to generate this timeseries,
+	// typically either the query, or the function stack used to transform
+	// specific results.
+	Specification string
+
+	// Metric tags.
+	Tags map[string]string
+
+	// options for graphing
+	GraphOptions map[string]string
+
+	// consolidationFunc specifies how the series will be consolidated when the
+	// number of data points in the series is more than the maximum number allowed.
```

Comment:
We can get rid of GraphOptions, this isn't relevant for a graphite only implementation.